server/channels/store/sqlstore/license_store.go

```diff
@@ -41,7 +43,7 @@ func (ls SqlLicenseStore) Save(license *model.LicenseRecord) (*model.LicenseReco
 		return nil, errors.Wrap(err, "license_tosql")
 	}
 	var storedLicense model.LicenseRecord
-	if err := ls.GetReplicaX().Get(&storedLicense, queryString, args...); err != nil {
+	if err := ls.GetMasterX().Get(&storedLicense, queryString, args...); err != nil {
```

Comment:
This change is likely unrelated to issue but here it seems just safer to read from the writer as well. I am assuming this is not an heavy used call but let me know if there are concerns.

Reply:
Ah, this actually should be an upsert call with no action if already exists. Can we change to that?

Reply:
Yeah, that makes sense, I'll review :+1:

Reply:
@agnivade Logic wise we expect to return the stored license (if present) so I don't think an upsert will work here cause it won't return anything. 

A few options:

- We keep it as is, both operations decoupled but running on the writer.
- We use a transaction to make both operations (fetch and insert) atomic.
- We only attempt the insert and on conflict we simply return the input license. It could work but it has some risks attached since we would be essentially trusting the input to be the same as what's stored in DB.

Reply:
I checked all instances of this call. Nowhere we are actually using the return value. I think we can safely change this to an `INSERT .. ON CONFLICT DO NOTHING` statement and not return anything.